tutorials/context-dependent-tool-tutorial.md

```diff
@@ -0,0 +1,375 @@
+## Intermediate Tool Tutorial: Context-dependent Tools
+
+Pre-req: [Creating a Simple Tool](../develop/spatial-tools/tutorial)
+
+Now that you know how to create a tool and interact with some basic RealityInterface APIs,
+we're going to explore some more interesting things you can do with tools. This tutorial gets a
+bit complicated, but as always please [use the forum](forum.spatialtoolbox.vuforia.com) if you
+have any questions.
+
+##### Setting Up
+
+First up, create a new directory in your edge server's  `addons/my-addon/tools` called
+`contextDependent`. This will sit alongside the `randomColor` directory from the previous
+tutorial. Copy-and-paste the same [icon.gif file](./images/icon.gif) as last time, and create an
+empty index.html file. The resulting directory tree will look like:
+
+![directory tree for new tool](./images/contextDependent-tool-directories.png)
+
+You'll need to enable this tool on the server frontend for it to become available to clients in
+the network. Go to the Spatial Tools view on localhost:8080 and click the On/Off button for the
+"contextDependent" tool so that it says **On**. It should appear in the pocket of your app if
+you're in the same network now, although you'll need to add some content to it first.
+
+In this tutorial, we'll look at a couple different ways that the contents of the tool can react
+to its spatial context.
+
+### Reacting to the position of the viewer
+
+One way that a tool can be spatially context-dependent is by responding to the relative location
+between it and its viewer – you!
+
+There are a few APIs that let a tool's contents react to its position. The most powerful one,
+which lets us extract all possible information about its relative location to the camera, is
+`addMatrixListener`. This is how a tool can subscribe to its **modelView matrix**. Don't panic
+– you don't need to know much about this matrix (or matrices in general). Just know that this
+is an array of 16 numbers (representing a 4x4 matrix of numbers) that fully describes where
+the tool is positioned in 3D space, relative to the camera. You can extract the scale of the
+tool from its first entry (`modelView[0]`), and the approximate distance to the camera from
+its 15th entry (`modelView[14]`).
+
+##### Demo 1: Adjust a continuous property based on distance
+
+In this first example, we continuously adjust the color of the tool as you move from about 0.5
+meters away from the tool to about 1 meter away from the tool:
+
+```javascript
+realityInterface.addMatrixListener(function(modelView, _projection) {
+  	// convert the scary matrix into easy-to-use values
```

Comment:
mixed tabs and spaces are making this look a bit weird

Reply:
good catch! converted them all to spaces.